java/code/src/com/redhat/rhn/frontend/action/systems/SPMigrationAction.java

```diff
@@ -225,7 +225,13 @@ public ActionForward execute(ActionMapping actionMapping,
                 logger.debug("Installed products are 'unknown'");
                 return forward;
             }
-            installedProducts.ifPresent(pset -> logger.debug(pset.toString()));
+            installedProducts.ifPresent(pset -> {
+                logger.debug(pset.toString());
```

Comment:
how does this solve the NPEx? looking at the exception in the Bugzilla, it seems it got caused by `pset.toString()`, but the new code still calls this method, even if the base product is `null`.

Reply:
I already created a previous PR to fix the NPE, but we tought it would be better to log a proper error in this case instead of just a debug log that displays an empty base product.

See `SUSEProductSet` line: 256
```
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("[base: ");
        if (baseProduct != null) {
            builder.append(baseProduct.getFriendlyName());
        }
```

Reply:
AHA! I was not aware of this. Makes sense. Thanks!

Reply:
Yeah sorry for the confusion :smiley: